File: 프로그래머스/lv2/68645. 삼각 달팽이/삼각 달팽이.py
def solution(n):
    answer = [[0]*(i+1) for i in range(n)]
    num = 1
    idx = 1
    row = -1
    col = 0
    
    while n>0:
        for _ in range(n):
            row += idx
            answer[row][col] = num
            num+=1
        n-=1
        for _ in range(n):
            col += idx
            answer[row][col] = num
            num+=1
        n-=1
        for _ in range(n):
            col -= idx
            row -= idx
            answer[row][col] = num
            num+=1
        n-=1
    return sum(answer, [])

File: 프로그래머스/lv2/68645. 삼각 달팽이/삼각 달팽이.js
function solution(n) {
    const answer = new Array(n);
    for (var i = 1; i <= n; i++) {
        answer[i-1] = new Array(i);
    }
    
    let num = 1, row = -1, col = 0
    console.log(num, row, col)
    
    return answer;
}